wagtail/wagtailcore/blocks/base.py

```diff
@@ -204,24 +207,100 @@ def get_prep_value(self, value):
         return value
 
     def get_context(self, value):
+        """
+        Return a dict of context variables (derived from the block value, or otherwise)
+        to be added to the template context when rendering this value through a template.
+        """
         return {
             'self': value,
             self.TEMPLATE_VAR: value,
         }
 
-    def render(self, value):
+    def _render_with_context(self, value, context=None):
+        """
+        Temporary hack to accommodate Block subclasses created for Wagtail <1.6 which
+        have overridden `render` with the type signature:
+            def render(self, value)
+        and will therefore fail when passed a `context` kwarg.
+
+        In Wagtail 1.8, when support for context-less `render` methods is dropped,
+        this method will be deleted (and calls to it replaced with a direct call to `render`).
+        """
+        try:
+            return self.render(value, context=context)
+        except TypeError:
```

Comment:
Hm. If 3-rd party block raises `TypeError` (I don't know why they may do that :)), `render` method will be called twice even if block have proper signature. It will be hard to debug for 3-rd party developers.

```python
    def render(self, value, context=None):
        # do something that raises a TypeError
```

Probably we need to use `inspect`. Something like this:
```python
import inspect


class Foo:
    def render(self, value):
        print("I'm alive")
    def _render_with_context(self, value, context=None):
        if 'context' in inspect.getargspec(self.render).args:
            return self.render(value, context=context)
        else:
            result = self.render(value)
            # Warn and bla-bla-bla


test = Foo()
test._render_with_context(1, {"two": 2})
```